src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -167,4 +167,61 @@ protected function instantiateController($class)
     {
         return new $class();
     }
+
+    private function getControllerError($callable)
+    {
+        if (is_string($callable)) {
+            if (!function_exists($callable)) {
+                return sprintf('Function "%s" does not exist.', $callable);
+            }
+
+            if (false !== strpos($callable, '::')) {
+                $callable = explode('::', $callable);
+            }
+        }
+
+        if (!is_array($callable)) {
+            return sprintf('Invalid type for controller given, expected string or array, got "%s".', gettype($callable));
+        }
+
+        if (2 !== count($callable)) {
+            return sprintf('Invalid format for controller, expected array(controller, method) or controller::method.');
+        }
+
+        list($controller, $method) = $callable;
+
+        if (is_string($controller) && !class_exists($controller)) {
```

Comment:
but the message will tell you ``Class "app.controller.typoed" does not exist.`` when you make a typo in your service id, which is not fine. We need an extension point to allow child classes to tweak the error message in this case to account for their additional features